connector/src/main/java/org/jboss/as/connector/subsystems/resourceadapters/ResourceAdapterSubsystemParser.java

```diff
@@ -369,7 +369,8 @@ private void writeConDef(XMLExtendedStreamWriter streamWriter, ModelNode conDef,
             streamWriter.writeEndElement();
         }
 
-        if (conDef.hasDefined(APPLICATION.getName()) || conDef.hasDefined(SECURITY_DOMAIN.getName())
+        if (conDef.hasDefined(AUTHENTICATION_CONTEXT.getName()) || conDef.has(AUTHENTICATION_CONTEXT_AND_APPLICATION.getName())
```

Comment:
@tadamski the problem is this:

```
conDef.has(AUTHENTICATION_CONTEXT_AND_APPLICATION.getName()
```

That 'has' needs to be 'hasDefined'.

Reply:
@bstansberry fixed